frontend/public/components/operator-lifecycle-manager/create-operand.tsx

```diff
@@ -107,40 +147,74 @@ export const CreateOperandForm: React.FC<CreateOperandFormProps> = (props) => {
   }, {} as FormValues);
 
   const [formValues, setFormValues] = React.useState<FormValues>({'metadata.name': 'example', 'metadata.labels': [], ...defaultFormValues, ...sampleFormValues});
-  const [error, setError] = React.useState();
+  const [error, setError] = React.useState<string>();
+  const [formErrors, setFormErrors] = React.useState<FieldErrors>({});
 
-  const updateFormValues = (values: FormValues) => (path: string, value: any) => {
-    return _.set(_.cloneDeep(values), path, value);
-  };
+  const updateFormValues = (values: FormValues) => (path: PropertyPath, value: any) => _.set(_.cloneDeep(values), path, value);
+
+  const submit = (event) => {
+    event.preventDefault();
+
+    const errors = fields.filter(f => !_.isNil(f.validation)).reduce((allErrors, field) => {
+      // NOTE: Can't use server-side error response due to missing info (https://github.com/kubernetes/kubernetes/issues/80718)
```

Comment:
Would much prefer to use the API server to return validation errors, but this bug prevents us from doing that. Should be fixed with CRDs using structural schemas.